src/inmanta/ast/type.py

```diff
@@ -199,6 +275,10 @@ def __eq__(self, other: object) -> bool:
             return NotImplemented
         return True
 
+    def has_custom_to_python(self) -> bool:
+        # All primitives can be trivially converted
+        return True
```

Comment:
This sounds risky. Wouldn't this cause e.g. primitive lists to be converted instead of being dynamic proxies? I find it tough to judge the impact, but I think this could have a real effect on (possibly rare) models, because object identity might change.

Reply:
works fine without